src/jiti.ts

```diff
@@ -77,12 +78,24 @@ export default function createJITI (_filename: string = process.cwd(), opts: JIT
     try { return nativeRequire.resolve(id, options) } catch (e) {}
   }
 
+  const _esmResolve = createESMResolver(_filename, { constraints: ['node'] })
```

Comment:
Since we are still in cjs context, would suggest adding `require` constraint.

Reply:
(as a workaround we can try/catch and resolve with `require` to hotfix estree issue but primarily prefer cjs to avoid unnecessary jiti transforms)